app/controllers/discussions_controller.rb

```diff
@@ -77,9 +78,19 @@ def unfavorite
     redirect_to discussions_path(discussion_id: params[:discussion_id], search: params[:search])
   end
 
+  def upvote
+    if current_user
+      @discussion = Discussion.find(params[:discussion_id])
+      @discussion.upvotes.create(user_id: current_user.id)
+      @discussion.score += 1
```

Comment:
Getting this error because discussion.score is default "nil": NoMethodError - undefined method `+' for nil:NilClass
So add a check for "if @discussion.score" to do += 1, else just set it equal to 1